static/app/views/settings/projectMetrics/utils/api.tsx

```diff
@@ -23,13 +23,27 @@ function filterTempIds(rules: MetricsExtractionRule[]) {
     })),
   }));
 }
+interface MetricRulesAPIOptions {
+  query?: string;
+}
 
-export const getMetricsExtractionRulesApiKey = (orgSlug: string, projectSlug: string) =>
-  [`/projects/${orgSlug}/${projectSlug}/metrics/extraction-rules/`] as const;
+export const getMetricsExtractionRulesApiKey = (
+  orgSlug: string,
+  projectSlug: string,
+  options?: MetricRulesAPIOptions
+) =>
+  [
+    `/projects/${orgSlug}/${projectSlug}/metrics/extraction-rules/`,
+    {query: {query: options?.query || undefined}},
+  ] as const;
```

Comment:
This will break optimistic updates and invalidation from `create`, `update` and `delete`, right?.

You can fix this by only passing the partial query key (endpoint url) when updating and invalidating.